core/commands/ls.go

```diff
@@ -212,20 +212,14 @@ The JSON output contains type information.
 }
 
 func makeDagNodeLinkResults(req *cmds.Request, dagnode ipld.Node) <-chan unixfs.LinkResult {
-	linkResults := make(chan unixfs.LinkResult)
-	go func() {
-		defer close(linkResults)
-		for _, l := range dagnode.Links() {
-			select {
-			case linkResults <- unixfs.LinkResult{
-				Link: l,
-				Err:  nil,
-			}:
-			case <-req.Context.Done():
-				return
-			}
+	links := dagnode.Links()
+	linkResults := make(chan unixfs.LinkResult, len(links))
+	for _, l := range links {
+		linkResults <- unixfs.LinkResult{
+			Link: l,
+			Err:  nil,
 		}
-	}()
+	}
```

Comment:
I believe you still need to close the channel and check for `req.Context.Done()`.

Reply:
sorry bout that bit. sloppy coding from moving fast.

I also don't know req.Context.Done() -- @magik6k do you know?

Reply:
went ahead and added the channel close.

I'm not sure we have to be super concerned about context.Done() because based on the way the channel is setup, the write is not a blocking write -- because the buffer for the channel is defined as being as long as the list of links. My understanding is that would be an issue if the request getting cancelled in a separate process could potentially leave a different go routine still writing cause it's doing a blocking write to a channel that is no longer being read from. In this case, we aren't starting a separate go routine and the channel writes are synchronous.

Then again, this code has no integration tests I think... I wonder if that's something that needs to be added.

Anyway, this function is really an outlier use case -- so if there's any confusion I can just revert to the original version.

Reply:
@hannahhoward did you push the change with the `close`.  I am not seeing it in the latest diff.

Let's keep this as unresolved until we either revert the change or figure out if we need to handle `context.Done()`.

Reply:
oops sorry failed to push even though I thought I had.